### PR TITLE
Merge empty_error

### PR DESCRIPTION
**Pull request by AdRiley**
### Pull Request Description

We don't need 3 copies of empty_error. This PR makes them one.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Array.enso [hunk 1]
```diff
@@ -707,7 +707,7 @@ type Array
 
              [0, 1, 2].to_array . reduce (+)
     reduce : (Any -> Any -> Any) -> Any -> Any
-    reduce self function ~if_empty=(Error.throw Empty_Error) = Vector.reduce self function if_empty
+    reduce self function ~if_empty=(Error.throw Empty_Error.Error Vector) = Vector.reduce self function if_empty
```
Comment (GregoryTravis) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Array.enso]:
```suggestion
    reduce self function ~if_empty=(Error.throw (Empty_Error.Error Vector)) = Vector.reduce self function if_empty
```

**Review by GregoryTravis — APPROVED** [review]

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Array.enso [hunk 2]
```diff
@@ -7,7 +7,7 @@ import project.Data.Pair.Pair
 import project.Data.Range.Range
 import project.Data.Sort_Direction.Sort_Direction
 import project.Data.Text.Text
-import project.Data.Vector.Empty_Error
+import Standard.Base.Errors.Empty_Error.Empty_Error
```
Comment (jdunkerley) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Array.enso]: Please use project within Standard.Base to refer to it.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Statistics.enso [hunk 3]
```diff
@@ -16,7 +16,7 @@ import project.Panic.Panic
 import project.Runtime.Ref.Ref
 from project.Data.Boolean import Boolean, False, True
 from project.Data.Range.Extensions import all
-from project.Data.Vector import Empty_Error
+import Standard.Base.Errors.Empty_Error.Empty_Error
```
Comment (jdunkerley) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Statistics.enso]: As above - project please.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Errors/Empty_Error.enso [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+## An error representing that the collection is empty.
+type Empty_Error
+    ## PRIVATE
+    Error (collection)
+
+    ## PRIVATE
+
+       Pretty prints the empty error.
+    to_display_text : Text
+    to_display_text self = "The " + self.collection.to_text + " is empty."
```
Comment (jdunkerley) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Errors/Empty_Error.enso]: Please add a new line at the end of the file.

**Review by jdunkerley — APPROVED** [review]
A few stylist in tweaks but looks good. And only one type makes me happy.

**Review by Akirathan — APPROVED** [review]

File: distribution/lib/Standard/Base/0.0.0-dev/src/Errors/Empty_Error.enso [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import project.Data.Text.Text
+
+## An error representing that the collection is empty.
+type Empty_Error
+    ## PRIVATE
+    Error (collection)
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Errors/Empty_Error.enso]:
```suggestion
    Error collection
```
I think the parentheses are not necessary here.

**Review by radeusgd — APPROVED** [review]

**Review by 4e6 — APPROVED** [review]

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Array.enso [hunk 6]
```diff
@@ -710,7 +710,7 @@ type Array
 
              [0, 1, 2].to_array . reduce (+)
     reduce : (Any -> Any -> Any) -> Any -> Any
-    reduce self function ~if_empty=(Error.throw Empty_Error) = Vector.reduce self function if_empty
+    reduce self function ~if_empty=(Error.throw (Empty_Error.Error Vector)) = Vector.reduce self function if_empty
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Array.enso:713]: Shouldn't this be
```suggestion
    reduce self function ~if_empty=(Error.throw (Empty_Error.Error Array)) = Vector.reduce self function if_empty
```

given that we are operating with an Array here??

I know they are pretty much interchangeable but they still are separate types.

(Sorry I just noticed it after getting a merge conflict with #8867)

Comment (AdRiley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Array.enso:713]: Yes. Good spot :) I will change that.

Comment (AdRiley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Array.enso:713]: https://github.com/enso-org/enso/pull/9019

**Pull request merged**